examples/experimental/dagster-airlift/dagster_airlift/core/load_defs.py

```diff
@@ -0,0 +1,249 @@
+    airflow_instance: AirflowInstance,
+) -> _SerializableDataFromDefs:
+    """Compute data from definitions that can be cached and used to construct the "final" definitions,
+    without calls to the airflow rest API.
+    """
+    downstreams_asset_dependency_graph: Dict[AssetKey, Set[AssetKey]] = defaultdict(set)
+    spec_transformations: Dict[AssetKey, AssetSpecTransformationData] = {}
+    all_asset_keys_per_dag_id: Dict[str, Set[AssetKey]] = defaultdict(set)
+    for asset in definitions.assets or []:
+        asset = check.inst(  # noqa: PLW2901
+            asset,
+            (AssetsDefinition, AssetSpec),
+            "Expected orchestrated defs to all be AssetsDefinitions or AssetSpecs.",
+        )
+        task_info = get_task_info_for_asset(airflow_instance, asset)
+        if task_info is None:
+            continue
+        migration_state_for_task = migration_state.get_migration_state_for_task(
+            dag_id=task_info.dag_id, task_id=task_info.task_id
+        )
+        specs = asset.specs if isinstance(asset, AssetsDefinition) else [asset]
+        for spec in specs:
+            spec_transformations[spec.key] = get_task_transformation_data(
+                task_info=task_info,
+                migration_state=migration_state_for_task,
+                spec=spec,
+            )
+            for dep in spec.deps:
+                downstreams_asset_dependency_graph[dep.asset_key].add(spec.key)
+            all_asset_keys_per_dag_id[task_info.dag_id].add(spec.key)
+    return _SerializableDataFromDefs(
+        spec_transformations=spec_transformations,
+        keys_in_dag=all_asset_keys_per_dag_id,
+        downstreams_asset_graph=downstreams_asset_dependency_graph,
+    )
+
+
+def construct_all_assets(
+    definitions: Definitions,
+    serialized_data: _SerializedData,
+) -> List[AssetsDefinition]:
+    return list(_transform_asset_specs(definitions, serialized_data.spec_transformations)) + list(
+        _construct_asset_specs(serialized_data.specs)
+    )
+
+
+def _transform_asset_specs(
+    definitions: Definitions, spec_transformations: Mapping[AssetKey, AssetSpecTransformationData]
+) -> Iterator[AssetsDefinition]:
+    """Apply asset spec transformations to the asset definitions."""
+
+    def _map_spec(spec: AssetSpec) -> AssetSpec:
+        if spec.key in spec_transformations:
+            return spec_transformations[spec.key].transform(spec)
+        return spec
+
+    for asset in definitions.assets or []:
+        asset = check.inst(  # noqa: PLW2901
+            asset,
+            (AssetSpec, AssetsDefinition),
+            "Expected orchestrated defs to all be AssetsDefinitions or AssetSpecs.",
+        )
+        assets_def = (
+            asset if isinstance(asset, AssetsDefinition) else external_asset_from_spec(asset)
+        )
+        yield assets_def.map_asset_specs(_map_spec)
+
+
+def _construct_asset_specs(
+    serializable_specs: Sequence[SerializableAssetSpec],
```

Comment:
slight preference for "AssetSpecData" or something because it is shorter and because variable name can be `spec_data`. `spec_data.to_asset_spec()` makes sense whereas `spec.to_asset_spec()` looks tautilogical

Reply:
agreed.